pkg/controllers/jobset_controller.go

```diff
@@ -196,30 +223,32 @@ func (r *JobSetReconciler) createJobs(ctx context.Context, js *jobset.JobSet, ow
 	ctx = ctrl.LoggerInto(ctx, log)
 
 	for _, rjob := range js.Spec.Jobs {
-		job, err := r.constructJobFromTemplate(js, &rjob)
+		jobs, err := r.constructJobsFromTemplate(js, &rjob)
```

Comment:
I find this a bit strange that we construct the full object of all jobs and then decide which ones to persist on the api server. I think we should have logic that decides which jobs to create (only job names is sufficient), and then for each we construct the object and create it on the api-server.

Reply:
Makes sense, I moved the `shouldCreateJob` call into `constructJobsFromTemplate` so we don't unnecessarily construct jobs we aren't going to create.